### PR TITLE
Rework cleanup logic

### PR DESCRIPTION
**Pull request by DevonianTeuchter**
Regenerate trap commands everytime a new command is added rather than
maintain a list (in file or pipe)
"Simplify" progress handler (remove duplicate code) and add logic to detect if orphaned

### CODE REVIEW

File: include/common.sh [hunk 1]
```diff
@@ -790,7 +792,7 @@ printError()
   printColors "${NC}${RED}${BOLD}***ERROR: ${NC}${RED}${1}${NC}" >&2
   [ -n "${xtrc}" ] && set -x
   mutexFree "zopen" # prevent lock from lingering around after an error
-  cleanupFunction
+cleanupFunction
```
Comment (IgorTodorovskiIBM) [inline include/common.sh:795]: Is this change intended?

Comment (DevonianTeuchter) [reply include/common.sh:795]: No, just me not spotting the indentation change as "show whitespace differences" wasn't enabled when I reinstated the cleanup call (as was testing without it)
